### ISSUE TITLE
Improve non-hold key callouts

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-Android#21]
### Terms

- [X] I have searched [open and closed design issues](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Adesign)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is to improve the key callouts that pop up when a key is pressed. As of now the callouts don't cover the key when they're pressed, with the user generally not seeing this as their thumb is covering the key, but the behavior is not correct based on comparisons to system keyboards.

Ideally there would be a full callout that covers the key, which then would transition to the hold callout if the user is holding a key to select alternate characters.

> [!NOTE]
> We also need to match the color of the key callout to that of the key.

### Contribution

This issue is thus of low priority as of now, but could be worked on later when the callouts are being investigated further. Others are welcome to contribute!

**Comment by Linfye** [comment]
@andrewtavis Does this issue still exist? Could you explain in details, like a screenshot?:)

**Comment by andrewtavis** [comment]
Will send something along later, @Linfye, but yes it does! :)

**Comment by andrewtavis** [comment]
In looking at it, there are a lot of issues with this functionality at this point, @Linfye, and it would be great if you could take a look at it :)

All of this is activated by `Show popup on keypress` in the settings, or at least the base popup. You can see that in dark mode the key popup is not colored appropriately:

![Screenshot 2024-11-30 at 15 23 04 (2)](https://github.com/user-attachments/assets/1ad3243a-e430-4a04-b179-44b1021b0489)

Beyond that is the original reason for this issue, which is that the callout there is a bit messy. It'd be great if the callout looks more like it does in the iOS app where it visually appears that the callout is coming from the key, not just hovering over it:

![Screenshot 2024-11-30 at 15 29 56 (2)](https://github.com/user-attachments/assets/f47aa06f-89dd-4b05-a1fa-76447f8996bf)

Beyond that, the hold key for alternate characters functionality seems to have been broken along the way, or isn't working well in dark mode. See the image below:

![Screenshot 2024-11-30 at 15 23 22 (2)](https://github.com/user-attachments/assets/974f84cf-fb2e-462f-af0a-91041ead3bc7)

The above is me holding the `A` key, but then there's no background behind the options to make it clear that if I swipe I can select one of the alternate `A` characters.

Let me know if you'd have interest in looking into any of the above! We could start with one PR to fix the color of the base callout in dark mode and make the hold to select callout visible again, and then we could do another PR to improve the callout itself 😊 Feel free to indicate that you have interest, and from there you can get started and I'll assign you later 🚀

**Comment by Linfye** [comment]
Yes, I'd like to work on this. Thanks for your explaination. :)

**Comment by andrewtavis** [comment]
Nice, @Linfye! Let us know if you have any questions :)

**Comment by andrewtavis** [comment]
Do you want to work on a fix for the long hold accent character selection background now, @Linfye? So a fix for the following above where the optional characters have no background:

![Screenshot 2024-11-30 at 15 23 22 (2)](https://github.com/user-attachments/assets/974f84cf-fb2e-462f-af0a-91041ead3bc7)

**Comment by Linfye** [comment]
Yes 😁

**Comment by Linfye** [comment]
Yes, but I've been quite busy lately. I plan to continue the scribe-related work after returning to college in March.

If anyone else is willing to take it over, they can do so. :)

**Comment by andrewtavis** [comment]
Sounds good, @Linfye! Looking forward to having you back, and all the best till then!

**Comment by andrewtavis** [comment]
Just added the following to the issue description above:

> [!NOTE]
> We also need to match the color of the key callout to that of the key.

Right now the color's a bit off :)

**Comment by Linfye** [comment]
I'm trying to replicate the effect of the iOS version, but it's quite challenging (I can't draw such a trapezoid very well). Maybe we can take inspiration from the current methods used by Gboard?

**Comment by andrewtavis** [comment]
Thanks for the clarifying questions, @Linfye :) All your deductions are correct, and the color for dark mode is the light mode background, I believe.

How is Gboard doing this?

**Comment by Linfye** [comment]
![Image](https://github.com/user-attachments/assets/df256d01-77fc-4265-8a0c-63434499b8cc)
![Image](https://github.com/user-attachments/assets/facfeaa7-9485-4437-a9a9-4fcd5a1d2d75)

As shown in the screenshots, I think this is an intuitive and convenient way.

**Comment by andrewtavis** [comment]
CC @angrezichatterbox on the potential design choices here :) Ideally we would have a similar design across the apps, but if it can't work I'm fine with going in a similar direction to Gboard 😊

**Comment by angrezichatterbox** [comment]
> CC [@angrezichatterbox](https://github.com/angrezichatterbox) on the potential design choices here :) Ideally we would have a similar design across the apps, but if it can't work I'm fine with going in a similar direction to Gboard 😊

It would be great to have a consistent design across both Android and iOS, but Android has traditionally never followed that approach. I'm not sure how well the implementation would work, even if we try to match it. Also, since Android users are more accustomed to the Gboard look, I feel it would be better to follow that direction. Like Apple seems to have trapezoid by default in their keyboards as well.

**Comment by andrewtavis** [comment]
For now I think that we can go with the circles to make sure that we get the coloration and other fixes in, and then from there we can maybe make another issue to bring the design to a similar place with iOS 😊 Doing this issue in a simple way will also prepare us for maybe doing a switch to trapezoids.

Feel free to proceed with the Gboard approach, @Linfye! Let us know if further questions come up :)

**Comment by andrewtavis** [comment]
We removed the shadows from the alternate/accent keys view and now are roughly at parity with the GBoard :) Thanks for the quick work to finalize this as a part of #511, @angrezichatterbox!
